### ISSUE TITLE
<Resource>.update method as alternative to retrieve -> update

### ISSUE THREAD

**Issue by mgroebner** [stripe/stripe-java#855]
It would be great to have an update method on resource classes directly. This would save one unnecessary request.

Example:

```
CustomerUpdateParams params = ...;
Customer.update(id, params);

```

**Comment by remi-stripe** [comment]
@cyrez Thanks for reaching out! This is definitely something we want to add to the library in the future as it's the only one left (out of 7 total) to require a `retrieve` and then a `save`.

I'm tagging this as future as we continue working on improvements to the library and can look into prioritizing that one.

**Comment by cyrez** [comment]
@remi-stripe Thanks for your fast reply, looking forward to see this addition. You guys did a great job improving the library the last few month, got much better to use.

**Comment by cyrez** [comment]
Just came across a similar problem and either a bug in the java lib or the docs:
[https://stripe.com/docs/api/customer_tax_ids/retrieve?lang=java](https://stripe.com/docs/api/customer_tax_ids/retrieve?lang=java)

Retrieving a TaxId is described in the docs with this:

```java
TaxId.retrieve("txi_123456789");
```

But currently is working like this:

```java
Customer customer = Customer.retrieve("cus_123");
customer.getTaxIds().retrive("txi_123456789");
```

I would prefer to call "retrieve" and "list" without retrieving the customer first.

**Comment by soutchay** [comment]
@remi-stripe I wanted to also weigh in on this issue. I've found that I would actually need to make 3 calls to create a Person
For example,
```
Account.retrieve().persons().create(params)
```
To avoid the extra calls we're doing the following: 
```
PersonCollectionCreateParams personCollectionCreateParams = PersonCollectionCreateParams.builder().setPersonToken("personToken").build();
PersonCollection personCollection = new PersonCollection();
personCollection.setUrl("/v1/accounts/" + accountId + "/persons");
personCollection.create(personCollectionCreateParams, requestOptions);
```
**Is it ok for us to be making the calls to the Stripe API in this way?**

**Comment by remi-stripe** [comment]
@soutchay I would not recommend doing it this way because we could make change to the infrastructure that would break your integration. You could, but you'd need to make sure that whenever you upgrade your version of stripe-java, even with a minor version, you triple check that we are not causing a regression for your code.

We are definitely aware of this limitation. We just shipped client/services in stripe-php 2 weeks ago and we do hope to revamp the stripe-java library in the future to make this easier too.

**Comment by soutchay** [comment]
Thank you for the quick response and recommendation! I look forward to the stripe-java library revamp

**Comment by pakrym-stripe** [comment]
This is now possible using the `StripeClient` type available starting in v23.1.1:

```java
StripeClient client = new StripeClient("api_key");

com.stripe.param.PlanUpdateParams params =
    com.stripe.param.PlanUpdateParams.builder().putMetadata("order_id", "6735").build();

com.stripe.model.Plan plan = client.plans().update("price_xxxxxxxxxxxxx", params);
```
